apps/qubit/modules/informationobject/actions/renameAction.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class InformationObjectRenameAction extends sfAction
+{
+  // allow modification of title, slug, and digital object filename
+  public function execute($request)
+  {
+    // return 401 if unauthorized
+    if (!sfContext::getInstance()->user->isAuthenticated())
```

Comment:
I think we should get the resource first and do a full ACL check (!QubitAcl::check($this->resource, 'update')).